### ISSUE TITLE
Make "convenience" code actions shown only when invoked manually

### ISSUE THREAD

**Issue by ulugbekna** [ocaml/ocaml-lsp#897]
Code actions that are proposed all the time such as "open related" and "type annotate" diminish the desire of a user to explore what actually useful code actions may be proposed by olsp (e.g., the proposed code actions are visually indicated using a lightbulb in vscode). 

I propose that we offer code actions that are convenient but don't relate to the current code action context only if the code action request was triggered manually rather than automatically (`reqParams.context.triggerKind = Invoked`).

A somewhat similar concern was voiced in #797 . 

What do people think? 

cc @rgrinberg

**Comment by rgrinberg** [comment]
> only if the code action request was triggered manually

How does one trigger the code action manually in vscode for example?

**Comment by ulugbekna** [comment]
Cmd+.

**Comment by rgrinberg** [comment]
I don't think this will affect my workflow, so I don't particularly mind. Let's hear some input from some other users.
